draft-ietf-tls-esni.md

```diff
@@ -686,72 +684,72 @@ Offering a GREASE extension is not considered offering an encrypted ClientHello
+  ECHConfig values of different versions. This allows a server to support
+  multiple versions at once.
 
 - The server MUST ignore all PSK identities in the ClientHello which correspond
   to ECH PSKs. ECH PSKs offered by the client are associated with the ECH
   name. The server was unable to decrypt then ECH name, so it should not resume
   them when using the plaintext SNI name. This restriction allows a client to
   reject resumptions in {{auth-public-name}}.
 
-Note that an unrecognized ClientEncryptedCH.config_id value may be a GREASE
-ECH extension (see {{grease-extensions}}), so it is necessary for servers to
-proceed with the connection and rely on the client to abort if ECH was required.
-In particular, the unrecognized value alone does not indicate a misconfigured
-ECH advertisement ({{misconfiguration}}). Instead, servers can measure
-occurrences of the "ech_required" alert to detect this case.
+Note that an unrecognized ClientECH.config_id value may be a GREASE ECH
+extension (see {{grease-extensions}}), so it is necessary for servers to proceed
+with the connection and rely on the client to abort if ECH was required. In
+particular, the unrecognized value alone does not indicate a misconfigured ECH
+advertisement ({{misconfiguration}}). Instead, servers can measure occurrences
+of the "ech_required" alert to detect this case.
 
-If the ClientEncryptedCH value matches a known ECHConfig, the server then
-decrypts ClientEncryptedCH.encrypted_ch, using the private key skR corresponding
+If the ClientECH value matches a known ECHConfig, the server then
+decrypts ClientECH.encrypted_ch, using the private key skR corresponding
 to ECHConfig, as follows:
 
 ~~~
-context = SetupBaseR(ClientEncryptedCH.enc, skR, "tls13-ech")
+context = SetupBaseR(ClientECH.enc, skR, "tls13-ech")
 
 ClientHelloInner =
-  context.Open("", ClientEncryptedCH.encrypted_ch)
+  context.Open("", ClientECH.encrypted_ch)
 
-ech_nonce_value = context.Export("tls13-ech-nonce", 16)
 ech_hrr_key = context.Export("tls13-ech-hrr-key", 16)
 ~~~
 
 If decryption fails, the server MUST abort the connection with a "decrypt_error"
-alert. Moreover, if there is no "ech_nonce" extension, or if its value does not
-match the derived ech_nonce, the server MUST abort the connection with a
-"decrypt_error" alert. Next, the server MUST scan ClientHelloInner for any
-"outer_extension" extensions and substitute their values with the values in
-ClientHelloOuter. It MUST first verify that the hash found in the extension
-matches the hash of the extension to be interpolated in and if it does not,
-abort the connections with a "decrypt_error" alert.
+alert. Next, the server MUST scan ClientHelloInner for any "outer_extension"
+extensions and substitute their values with the values in ClientHelloOuter. It
+MUST first verify that the hash found in the extension matches the hash of the
+extension to be interpolated in and if it does not, abort the connections with a
+"decrypt_error" alert.
```

Comment:
Should the server also assert that the random values are different?

Reply:
What would be the purpose?

Reply:
(Chatted offline -- maybe not now!)